src/ucp/rndv/rndv.c

```diff
@@ -1631,24 +1630,30 @@ static ucs_status_t ucp_rndv_send_start_put_pipeline(ucp_request_t *sreq,
     }
 
     ucp_request_send_state_init(fsreq, ucp_dt_make_contig(1), 0);
-    fsreq->super_req                    = sreq;
-    fsreq->flags                        = 0;
-    fsreq->send.buffer                  = UCS_PTR_BYTE_OFFSET(sreq->send.buffer,
-                                                              rndv_base_offset);
-    fsreq->send.length                  = rndv_size;
-    fsreq->send.mem_type                = sreq->send.mem_type;
-    fsreq->send.ep                      = sreq->send.ep;
-    fsreq->send.lane                    = sreq->send.lane;
-    fsreq->send.rndv_put.rkey           = sreq->send.rndv_put.rkey;
-    fsreq->send.rndv_put.uct_rkey       = sreq->send.rndv_put.uct_rkey;
-    fsreq->send.rndv_put.rreq_remote_id = rndv_rtr_hdr->rreq_id;
-    fsreq->send.rndv_put.remote_address = rndv_rtr_hdr->address;
-    fsreq->send.state.dt.offset         = 0;
+    fsreq->super_req                = sreq;
+    fsreq->flags                    = 0;
+    fsreq->send.buffer              = UCS_PTR_BYTE_OFFSET(sreq->send.buffer,
+                                                          rndv_base_offset);
+    fsreq->send.length              = rndv_size;
+    fsreq->send.mem_type            = sreq->send.mem_type;
```

Comment:
maybe define a common initializer func to reduce code duplication?

Reply:
done